src/lib/attach-listener.ts

```diff
@@ -20,11 +20,29 @@ const bindMessageHandler = (
   optionsRef: MutableRefObject<Options>,
   setLastMessage: Setters['setLastMessage'],
 ) => {
+  let heartbeatCb: () => void;
+
+  if (optionsRef.current.heartbeat && webSocketInstance instanceof WebSocket) {
+    const heartbeatOptions =
+      typeof optionsRef.current.heartbeat === "boolean"
+        ? undefined
+        : optionsRef.current.heartbeat;
+    heartbeatCb = heartbeat(webSocketInstance, heartbeatOptions);
+  }
+
   webSocketInstance.onmessage = (message: WebSocketEventMap['message']) => {
+    heartbeatCb?.();
     optionsRef.current.onMessage && optionsRef.current.onMessage(message);
     if (typeof optionsRef.current.filter === 'function' && optionsRef.current.filter(message) !== true) {
       return;
     }
+    if (
+      optionsRef.current.heartbeat &&
+      typeof optionsRef.current.heartbeat !== "boolean" &&
+      optionsRef.current.heartbeat?.kind === message.data
```

Comment:
If the heartbeat implementation is made with a ping/pong strategy, then this will not ignore the "pong" from the server.

Reply:
That makes sense -- I think I misunderstood the added documentation to imply that they need to match:

```
The `kind` of the ping message can be changed to any string, but it must be the same on the server side.
```